### PR TITLE
Add arrow buttons on either side of the title on IterationView to go to previous/next iteration

### LINKED ISSUE #291: [FR] Add arrow buttons on either side of the title on IterationView to go to previous/next iteration
In [an iteration view](https://www.bugzy.org/iteration/123.3), the title displays the iteration name. On the left side of this title, there could be a button leading to the previous iteration (in this case 123.2) and to the right, a button leading to the next iteration (124.1). Just simple left/right arrows would do it.

### PR DESCRIPTION
**Pull request by aminomancer**
Fixes #291

### CODE REVIEW

**Comment by punamdahiya** [comment]
Looks like some patches are committed directly to master which conflicts and this PR needs to be rebased with latest master

**Comment by aminomancer** [comment]
Rebased, should apply cleanly now

File: src/content/components/IterationView/IterationView.tsx [hunk 1]
```diff
@@ -227,8 +226,46 @@ export const IterationView: React.FunctionComponent<
     },
     [metas]
   );
+  const heading = React.useMemo(() => {
+    const nextIteration = iterations.getAdjacentIteration(1, props.iteration);
```
Comment (punamdahiya) [inline src/content/components/IterationView/IterationView.tsx:230]: For current iteration view, we can hide next arrow as it's mostly empty and if really needed can be reached from select iteration drop down.
```

 const nextIteration =
      props.iteration !== props.currentIteration.number
        ? iterations.getAdjacentIteration(1, props.iteration)
        : null;
```

File: src/content/components/Router/Router.js [hunk 2]
```diff
@@ -508,6 +508,11 @@ export class Router extends React.PureComponent {
               <Route exact={true} path="/">
```
Comment (punamdahiya) [inline src/content/components/Router/Router.js:508]: With easily accessible arrows and Select iteration drop down, we can remove `Next Iteration` menu item in left nav (Line 163 above)

**Review by punamdahiya — APPROVED** [review]
Looks great! Left feedback on further optimizing next iteration flow. Thanks for the patch!

File: src/content/components/Router/Router.js (hunk 2 above)
Comment (aminomancer) [reply src/content/components/Router/Router.js:508]: Good idea. But if we remove the next arrow from the current iteration view, then there won't be a way to reach Next Iteration except from the dropdown picker. So do you still want to do both? Remove the next arrow from current iteration _and_ remove `Next Iteration` menu item? Might make sense to just remove one or the other

Comment (punamdahiya) [reply src/content/components/Router/Router.js:508]: IMO `Next Iteration` view has usually less to no bugs and having access from select dropdown should suffice.

**Comment by aminomancer** [comment]
Great, thanks! Resolved feedback.

**Pull request merged**